mesos.yml

```diff
@@ -0,0 +1,310 @@
+        service nginx restart
+        
+        PUBLIC_IP=$(curl http://169.254.169.254/latest/meta-data/public-ipv4)
+        
+        wget https://github.com/mesosphere/mesos-dns/releases/download/v0.5.2/mesos-dns-v0.5.2-linux-amd64
+        mv mesos-dns-v0.5.2-linux-amd64 /usr/local/sbin/mesos-dns
+        chmod +x /usr/local/sbin/mesos-dns 
+        
+        cat <<EOT >> /etc/mesos-dns-config.json
+        {
+          "zk": "zk://$IPVAR1:2181/mesos",
+          "masters": ["$IPVAR1:5050"],
+          "refreshSeconds": 60,
+          "ttl": 60,
+          "domain": "mesos",
+          "port": 53,
+          "resolvers": ["8.8.8.8"],
+          "timeout": 5, 
+          "httpon": true,
+          "dnson": true,
+          "httpport": 8123,
+          "externalon": true,
+          "listener": "0.0.0.0",
+          "SOAMname": "ns1.mesos",
+          "SOARname": "root.ns1.mesos",
+          "SOARefresh": 60,
+          "SOARetry":   600,
+          "SOAExpire":  86400,
+          "SOAMinttl": 60,
+          "IPSources": ["mesos", "host"]
+        }
+        EOT
+        mesos-dns -config=/etc/mesos-dns-config.json & 
+        sed -i "s/nameserver 10.0.0.2/nameserver $IPVAR1\nnameserver 10.0.0.2/" /etc/resolv.conf
+        echo "You can access the Marathon Web UI at http://${var.koding_user_username}.mesosphere.live"
+        echo ""
+        echo "You can access the Mesos Web UI at http://${var.koding_user_username}-mesos.mesosphere.live"
+        
+        
+    slave_1:
+      instance_type: m4.2xlarge
+      ami: ''
+      root_block_device:
+        volume_size: 50
+      tags:
+        Name: '${var.koding_user_username}-${var.koding_group_slug}'
+      user_data: |-
+        export DEBIAN_FRONTEND=noninteractive
+        #setting the new source
+        apt-key adv --keyserver keyserver.ubuntu.com --recv E56151BF
+        DISTRO=$(lsb_release -is | tr '[:upper:]' '[:lower:]')
+        CODENAME=$(lsb_release -cs)
+        echo "deb http://repos.mesosphere.io/$DISTRO $CODENAME main" | sudo tee 	/etc/apt/sources.list.d/mesosphere.list
+        add-apt-repository ppa:openjdk-r/ppa
+        
+        #Updating packages
+        apt-get update -y
+        sleep 10
+        apt-get install -y openjdk-8-jdk
+        sleep 10
+        update-alternatives --config java
+        sleep 10
+        update-alternatives --config javac
+        sleep 10
+        #Installing mesos
+        sudo apt-get install -y mesos
+        sleep 10
+        #Getting the IP add
+        IPVAR1=${aws_instance.master.private_ip}
+        IPVAR=`ip addr | grep -Po '(?!(inet 127.\d.\d.1))(inet \K(\d{1,3}\.){3}\d{1,3})'`
+        
+        stop zookeeper
+        sleep 10
+        echo manual | sudo tee /etc/init/zookeeper.override
+        stop mesos-master
+        sleep 10
+        manual | sudo tee /etc/init/mesos-master.override
+        
+        #modifying the Zookepper mesos file
+        echo "zk://$IPVAR1:2181/mesos" > /etc/mesos/zk
```

Comment:
Using a private IP for a slave would case any API that hits slave not work from Marathon UI, e.g. link details.

![pic](http://g.recordit.co/Q0e4LjNKo7.gif)